utils/hooks/useHasPdf.ts

```diff
@@ -4,41 +4,54 @@ import useStore from "@/store";
 
 const cache: Record<string, boolean> = {};
 
-const soPdfEndpoint = process.env.NEXT_PUBLIC_SO_PDF_ENDPOINT;
-
-const useHasPdf = (forestType: null | string, url = `${soPdfEndpoint}`) => {
-  const activeProfile = useStore((state) => state.activeProfile);
+/**
+ * Custom hook to check if a PDF exists at the given URL.
+ * It caches the result to avoid repeated network requests.
+ *
+ * @param {string | null} pdfUrl - The URL of the PDF to check.
+ * @returns {boolean} - True if the PDF exists, false otherwise.
+ */
+const useHasPdf = (pdfUrl: null | string, profiles: string[] = []) => {
   const [hasPdf, setHasPdf] = useState(false);
+  const activeProfile = useStore((state) => state.activeProfile);
+  const profilesString = profiles.toString();
+
   useEffect(() => {
+    if (!pdfUrl || (profiles.length && !profiles.includes(activeProfile))) {
+      setHasPdf(false);
+      return;
+    }
     let abortCtrl = new AbortController();
-    if (forestType && activeProfile === "so") {
-      const pdfUrl = `${url}/${forestType.replace("*", "stern")}.pdf`;
```

Comment:
How about keeping this `replace` logic encapsulated inside this hook? Or move it to a helper function like `getTreeTypePdfUrl`?